kernel/thread.c

```diff
@@ -519,9 +519,14 @@ void thread_resched(void) {
     if (thread_is_idle(oldthread)) {
         lk_bigtime_t now = current_time_hires();
         thread_stats[cpu].idle_time += now - thread_stats[cpu].last_idle_timestamp;
+    } else {
+        oldthread->stats.total_run_time += current_time_hires() - oldthread->stats.last_run_timestamp;
```

Comment:
Suggestion: since every path through these two if blocks now involves a call to current_time_hires() it'll probably be cleaner to grab a snapshot into a variable `lk_bigtime_t now` before the ifs and just use it inside the loop. Should be the same codegen either way really.

Reply:
Will do.

Reply:
Done in latest version.